sdk/eventhubs/azure-messaging-eventhubs/migration-guide.md

```diff
@@ -1,205 +1,170 @@
-- [Migrating code from `EventHubClient` to `EventHubProducerAsyncClient` for sending events using automatic routing](#migrating-code-from-eventhubclient-to-eventhubproducerasyncclient-for-sending-events-using-automatic-routing)
-- [Migrating code from `EventHubClient` to `EventHubProducerAsyncClient` for sending events with partition key](#migrating-code-from-eventhubclient-to-eventhubproducerasyncclient-for-sending-events-with-partition-key)
-- [Migrating code from `PartitionReceiver` to `EventHubConsumerAsyncClient` for receiving events in batches](#migrating-code-from-partitionreceiver-to-eventhubconsumerasyncclient-for-receiving-events-in-batches)
-- [Migrating code from `EventProcessorHost` to `EventProcessorClient` for receiving events](#migrating-code-from-eventprocessorhost-to-eventprocessorclient-for-receiving-events)
+In v5:
 
-### Migrating code from `PartitionSender` to `EventHubProducerAsyncClient` for sending events to a partition
+```java
+public static void main(String[] args) throws Exception {
+    // This builds an async client. The sync version can be created by
+    // calling buildProducerClient() to create an EventHubProducerClient.
+    EventHubProducerAsyncClient client = new EventHubClientBuilder()
+        .connectionString("connection-string-for-an-event-hub")
+        .buildAsyncProducerClient();
 
-In v3, events could be published to a single partition using `PartitionSender`.
+    // Do things with the client.
+    // Dispose of client.
+    client.close();
+  }
+```
 
-In v5, this has been consolidated into a more efficient `send(EventDataBatch)` method. Batching merges information from
-multiple events into a single sent message, reducing the amount of network communication needed vs sending events one at
-a time. Events are published to a specific partition [`CreateBatchOptions.setPartitionId()`][CreateBatchOptions] is set
-before calling `createBatch(CreateBatchOptions)`.
+### Send events to an Event Hub
 
-The code below assumes all events fit into a single batch. For a more complete example, see sample: [Publishing events
-to specific partition][PublishEventsToSpecificPartition].
+In v3, events could be published to a single partition using `PartitionSender` or automatically routed to an available partition via `EventHubClient`.
 
-So in v3:
-```java
-EventHubClient client = EventHubClient.createFromConnectionStringSync(
-    "connection-string-for-an-event-hub", Executors.newScheduledThreadPool(4));
-List<EventData> events = Arrays.asList(EventData.create("foo".getBytes()));
+In v5, the **same client** can create an [EventDataBatchs][EventDataBatch] that routes to a specific partition or use automatic routing. Events are published to a specific partition using [`CreateBatchOptions.setPartitionId()`][CreateBatchOptions]; otherwise, the batch is automatically routed.
```

Comment:
Since in v3 there was only a single client ever using the "same client" does not drive home the point that we use the same class, but different overloads. How about the below?

```suggestion
In v5, the option to send single events is dropped to encourage sending events in batches for better throughput. 
You can send a list of events or create and send an event batch as before. Instead of using different classes, sending to a specific partition and making use of the automatic routing is done using overloads of the same methods. For example:
- Passing a list of events to the `send` method with no options makes use of automatic routing. If `partitionId` is passed in the options to this method, the events are sent to the given partition
- Creating a batch of events using the `createBatch` method without setting the `partitionId` in the options makes use of automatic routing. If `partitionId` is passed in the options to this method, the events are sent to the given partition
```